cmd/openshift-tuned.go

```diff
@@ -813,61 +547,51 @@ func (resyncPeriod *resync) changeWatcher() (err error) {
 
 			if string(data) == "stop" {
 				if err := tunedStop(&s); err != nil {
-					glog.Errorf("%s", err.Error())
+					klog.Errorf("%s", err.Error())
 				}
 				return nil
 			}
 
 		case <-tunedExit:
 			cmd = nil // cmd.Start() cannot be used more than once
-			return fmt.Errorf("Tuned process exitted")
+			return fmt.Errorf("tuned process exitted")
 
 		case fsEvent := <-wFs.Events:
-			glog.V(2).Infof("fsEvent")
+			klog.V(2).Infof("fsEvent")
 			// Ignore Write and Create events, wait for the removal of the old ConfigMap to trigger reload
 			if fsEvent.Op&fsnotify.Remove == fsnotify.Remove {
-				glog.V(1).Infof("Remove event on: %s", fsEvent.Name)
+				klog.V(1).Infof("remove event on: %s", fsEvent.Name)
 				tuned.change.cfg = true
 			}
 
 		case err := <-wFs.Errors:
-			return fmt.Errorf("Error watching filesystem: %v", err)
-
-		case podEvent, ok := <-wPod.ResultChan():
-			glog.V(2).Infof("wPod.ResultChan()")
-			if !ok {
-				return fmt.Errorf("Pod event watch channel closed.")
-			}
-			podChangeHandler(podEvent, &tuned)
-
-		case <-tickerPull.C:
-			glog.V(2).Infof("tickerPull.C")
-			if err := pullLabels(clientset, &tuned, nodeName); err != nil {
-				return err
-			}
-			if resyncPeriod.now/2 >= resyncPeriod.min {
-				// we've increased the original resyncPeriod due to errors; there was a successful
-				// pull now, converge back to the original resyncPeriod.min value
-				resyncPeriod.now /= 2
-				glog.V(1).Infof("Lowering resyncPeriod to %d", resyncPeriod.now)
-				tickerPull = time.NewTicker(time.Second * time.Duration(resyncPeriod.now))
-			}
+			return fmt.Errorf("error watching filesystem: %v", err)
 
 		case <-tickerReload.C:
-			glog.V(2).Infof("tickerReload.C")
+			klog.V(2).Infof("tickerReload.C")
 			if err := timedTunedReloader(&tuned); err != nil {
 				return err
 			}
 		}
 	}
+
+	return nil
 }
 
 func retryLoop() (err error) {
-	var resyncPeriod resync
-	resyncPeriod.now = pullResyncPeriodWithJitter()
-	resyncPeriod.min = resyncPeriod.now
+	const (
+		errsMax        = 5  // the maximum number of consecutive errors within errsMaxWithinSeconds
+		sleepRetryInit = 10 // the initial retry period [s]
+	)
+	var (
+		errs       int
+		sleepRetry int64 = sleepRetryInit
+		// sum of the series: S_n = x(1)*(q^n-1)/(q-1) + add 60s for each changeWatcher() call
+		errsMaxWithinSeconds int64 = (sleepRetry*int64(math.Pow(2, errsMax)) - sleepRetry) + errsMax*60
```

Comment:
What is the variable `errsMaxWithinSeconds`?

Reply:
It is a duration in seconds.  In particular the sum of the geometrical series of `errsMax` sleepRetry(s) + some extra time for each retry.  `sleepRetry` in increased 2x for every failure.  With the current code (unlike prior to the rearchitecture), I have not seen *any* retries, but I'd say it still make some sense to have a retry function of some sort.